File: src/main/java/seremis/geninfusion/soul/gene/newAI/restrictSun/GeneAIRestrictSun.java
package seremis.geninfusion.soul.gene.newAI.restrictSun;

import seremis.geninfusion.api.soul.IAllele;
import seremis.geninfusion.api.soul.lib.Genes;
import seremis.geninfusion.soul.MasterGene;
import seremis.geninfusion.soul.allele.AlleleBoolean;

public class GeneAIRestrictSun extends MasterGene {

    public GeneAIRestrictSun() {
        addControlledGene(Genes.GENE_AI_RESTRICT_SUN_INDEX);
    }

    @Override
    public Class<? extends IAllele> possibleAlleles() {
        return AlleleBoolean.class;
    }
}

File: src/main/java/seremis/geninfusion/soul/gene/newAI/GeneAIRestrictSun.scala
package seremis.geninfusion.soul.gene.newAI

import seremis.geninfusion.api.soul.lib.Genes
import seremis.geninfusion.soul.MasterGene

class GeneAIRestrictSun extends MasterGene(classOf[Boolean]) {
    addControlledGene(Genes.GeneAIRestrictSunIndex)
}
